### DISCUSSION TITLE
grpcurl client hangs when grpc request is blocked

### DISCUSSION THREAD

**Discussion by dspeg** [corazawaf/coraza#877] [Q&A]
I'm running a previous build of Coraza proxy wasm on an Istio sidecar proxy. I have configured a SecLang rule to deny requests containing a test pattern in request headers.  

Trying to know how the WAF handles grpc requests, I sent a request from "grpcurl" with a header that contains the test pattern. Then the "grpcurl" client hangs.  What could be the root cause?  Thanks! 

CC @jcchavezs @M4tteoP

**Comment by M4tteoP** [comment]
Hey @dspeg! Is the client hanging only when the test pattern is matched and the request should be denied, or it happens for any request? Would you be able to reproduce the issue directly against an Envoy instance (like tweaking the [example](https://github.com/corazawaf/coraza-proxy-wasm/tree/main/example) and providing us an example of request (alongside the tweaked example))?

**Comment by dspeg** [comment]
Thanks for following up! The client hangs only when the test pattern in header is matched and the request should be denied.

Have grpc requests been tested on more recent Coraza Proxy Wasm builds?

I am working on other tasks right now. In a few days I will get some logs and provide examples as you suggested.

**Comment by M4tteoP** [comment]
> The client hangs only when the test pattern in header is matched and the request should be denied.

Great, it already is a good point
> Have grpc requests been tested on more recent Coraza Proxy Wasm builds?

Not that I am aware of
> I am working on other tasks right now. In a few days I will get some logs and provide examples as you suggested.

Thanks! I might be able to also give you soon also a patched wasm to try out a possible fix

**Comment by dspeg** [comment]
Thanks a lot!

**Comment by M4tteoP** [comment]
Hey, as promised, this is a patched wasm that might fix the issue related to triggering an interruption with grpc.
Direct OCI image: `oci://docker.io/deltag/coraza-proxy-wasm:grpc_int`.
Branch: https://github.com/M4tteoP/coraza-proxy-wasm/tree/grpc_response
Please, consider this image/branch just for testing purposes. Thanks for any feedback and updates about this issue!

**Reply by M4tteoP** [reply]
Hi @dspeg, by any chance did you give it a go? Do you have any feedback? Thanks!
